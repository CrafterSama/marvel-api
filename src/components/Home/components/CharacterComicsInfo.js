import React from 'react'
import { CharacterModal, CharacterModalName, CharacterDescription, ComicList, ComicListItem } from '../../../styles'


const CharacterComicsInfo = ({ data }) => {
  return (
    <CharacterModal>
      <CharacterModalName>{data.name}</CharacterModalName>
      <CharacterDescription>{data.description}</CharacterDescription>
      <ComicList>
        {data.comics.items.map((comic, key) => (
          <ComicListItem key={key}>{comic.name}</ComicListItem>
        ))}
      </ComicList>
    </CharacterModal>
  );
}

export default CharacterComicsInfo
