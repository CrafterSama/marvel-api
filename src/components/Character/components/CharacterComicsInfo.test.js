import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import CharacterComicsInfo from './CharacterComicsInfo';

configure({ adapter: new Adapter() });

describe('Test for <CharacterComicsInfo /> Component', () => {

  let data = {};
  let toggle = () => {};

  test('matches snapshot', () => {
    const wrapper = shallow(
      <CharacterComicsInfo
        data={data}
        toggle={toggle}
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
