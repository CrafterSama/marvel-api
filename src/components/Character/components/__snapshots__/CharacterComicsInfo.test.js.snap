// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <CharacterComicsInfo /> Component matches snapshot 1`] = `
<styles__CharacterModal>
  <styles__CharacterModalName />
  <styles__CharacterDescription>
    No Description
  </styles__CharacterDescription>
  <styles__ComicList>
    <h4>
      Comics:
    </h4>
    <p>
      No Comics Associated
    </p>
  </styles__ComicList>
</styles__CharacterModal>
`;
